react/features/recording/middleware.ts

```diff
@@ -390,3 +399,27 @@ function _showRecordingErrorNotification(session: any, dispatch: IStore['dispatc
         APP.API.notifyRecordingStatusChanged(false, mode, error, isRecorderTranscriptionsRunning(getState()));
     }
 }
+
+/**
+ * Mutes audio and video and displays the RecordingConsentDialog when the conditions are met.
+ *
+ * @param {any} recorderSession - The recording session.
+ * @param {Function} dispatch - The Redux dispatch function.
+ * @param {Function} getState - The Redux getState function.
+ * @returns {void}
+ */
+function _showExplicitConsentDialog(recorderSession: any, dispatch: IStore['dispatch'], getState: IStore['getState']) {
+    const { requireRecordingConsent } = getState()['features/dynamic-branding'];
+
+    if (!requireRecordingConsent
+        && (recorderSession._statusFromJicofo === JitsiRecordingConstants.status.OFF
```

Comment:
Is there not better way to check? Maybe we need a helper to encapsulate these weird checks?